Add Features & Analysis to Secondary Columns

mykeels opened the pull request:
Resolves #5701

**Overall change:** 
_Add a FeaturesAnalysis container, which uses fixtures data to render a Features & Analysis section._

**Code changes:**
- _Added a FeaturesAnalysis container._
- _Added sample fixtures data files (LTR and RTL)._
- _Added stories and unit tests._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

Commit:
feat: first draft (wip) with fixture data

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
fix: specify the right aria-labelledby

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

mykeels commented:
![image](https://user-images.githubusercontent.com/11996508/75784499-2eb0cd00-5d5a-11ea-9e4c-4525241dac70.png)

Note the "Features Analysis" section on the right. You will notice the title has a white background. This is because the `SectionLabel` component used, is the same as used in this related content, just before the footer:

![image](https://user-images.githubusercontent.com/11996508/75783280-396a6280-5d58-11ea-91a2-bf580da4af77.png)

We'll need a way to pass the right style to it.

Commit:
fix: rm timestamp from features-analysis demo data

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
fix: change MAP type to STY

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

greenc05 commented on src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,317 @@
+exports[`CpsRelatedContent should render Story Feature components in Live environment 1`] = `
+<DocumentFragment>
+  <section
+    aria-labelledby="related-content-heading"
```
incorrect name for this "related-content-heading", should be "features-analysis-heading"?

greenc05 replied:
I see you've update it now.... thanks

greenc05 commented on src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,293 @@
+                    class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
+                  >
+                    <div
+                      class="lazyload-placeholder"
```
should we have an example of an image with alt text??

mykeels replied:
This resolves to something like:
```html
<img 
srcset="https://ichef.bbci.co.uk/news/70/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 70w, https://ichef.bbci.co.uk/news/95/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 95w, https://ichef.bbci.co.uk/news/144/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 144w, https://ichef.bbci.co.uk/news/183/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 183w, https://ichef.bbci.co.uk/news/240/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 240w, https://ichef.bbci.co.uk/news/320/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 320w, https://ichef.bbci.co.uk/news/624/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg 624w" 
alt="Some people at the NTA awards" 
src="https://ichef.bbci.co.uk/news/660/cpsdevpb/6ACE/test/_63724372_gettyimages-1098075358.jpg" 
sizes="(max-width: 1008px) 33vw, 237px" 
width="976" 
class="StyledImg-sc-7vx2mr-0 hIxkbt">
```

greenc05 commented:
Will this have a grey background as per your screenshot? https://github.com/bbc/simorgh/pull/5751#issuecomment-593971157

greenc05 commented on src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,293 @@
+                class="TextGridItem-sc-1lkbq5i-0 jJWtYv"
+                dir="ltr"
+              >
+                <h3
```
Just realised that we shouldn't have headings (h3's) for each of the promos, as there is no related content below them and this fails an accessibility standard. Is it possible to change this to a div/span please?

mykeels replied:
Yes, this has been overridden to make it a DIV.

jamesdonoh replied:
@greenc05 the use of h3 appears to be an existing problem with the Related Content component which the work here is heavily based on (e.g. see https://www.bbc.com/pidgin/tori-51728785).

We've just had a discussion and are going to create a separate issue to look at an accessible solution for both Related Content and Features and Analysis in the round, rather than applying an overly-specific fix here.

mykeels replied:
See https://github.com/bbc/simorgh/issues/5793

greenc05 replied:
Ok, so apologies if my use of the words 'related content' confused things here. If there is no text content below the promo title, then the title should not be a heading, as headings should have related content below them. So in the case of F&A, each li would not have any content below the 'title'. However in the 'Related content' component we have the time stamp below the promo title/heading so technically this is fine as a heading https://github.com/bbc/bbc-a11y/blob/master/lib/standards/tests/contentMustFollowHeadings.js

greenc05 replied:
The only other exception that I'm aware of is the onwards journey component in the middle of the article/story body text, this component cannot have any headings at all due to being in the article/story body and would head up all the article/story content below it, which would not be accurate

Commit:
Merge branch 'latest' into add-features-analysis

mykeels commented:
https://github.com/bbc/simorgh/pull/5751#issuecomment-594024646

@greenc05 I thought they'd all have grey backgrounds. I now see that might be because the sections with grey backgrounds are placeholders. Is that what's happening?

Commit:
Merge branch 'latest' into add-features-analysis

greenc05 commented:
> [#5751 (comment)](https://github.com/bbc/simorgh/pull/5751#issuecomment-594024646)
> 
> @greenc05 I thought they'd all have grey backgrounds. I now see that might be because the sections with grey backgrounds are placeholders. Is that what's happening?

I'm not saying they shouldn't have grey backgrounds, I don't know if they should or not, I was just checking.... could you tell me the hex value of the grey please?

mykeels commented:
https://github.com/bbc/simorgh/pull/5751#issuecomment-594402076

```
#ECEAE7
```

Commit:
Merge branch 'latest' into add-features-analysis

greenc05 commented:
Thanks for the colour. FYI there are some colour contrast issues, I've asked UX for a solution

Commit:
fix: rm unused vars

Commit:
fix: target headline span as transparent

Commit:
feat: override default StoryPromo headline

Commit:
Merge branch 'latest' into add-features-analysis

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,99 @@
+  }
+`;
+
+const FeaturesAnalysis = ({ content, enableGridWrapper }) => {
```
Do we need `enableGridWrapper` for this component? I think it was only needed for `CpsRelatedContent` because it was integrated differently in `MediaAsset` pages compared to `Story` pages.

mykeels replied:
What does the grid wrapper do?

mykeels replied:
Seen.

I've taken it out.

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,99 @@
+    );
+  FeaturesAnalysisWrapper.propTypes = {
+    children: node.isRequired,
+  };
```
Do we need propTypes on internal components?

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.test.jsx:
```diff
@@ -0,0 +1,62 @@
+
+  it('should render Story Feature components in Live environment', () => {
+    const { asFragment } = renderFeaturesAnalysis({
+      bbcOrigin: 'https://www.bbc.co.uk',
```
How does the change of origin make this and the test above behave differently?

mykeels replied:
it doesn't.

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.test.jsx:
```diff
@@ -0,0 +1,62 @@
+
+  it('should have a "region" role (a11y)', () => {
+    renderFeaturesAnalysis();
+    expect(document.querySelectorAll(`[role='region']`).length).toBe(1);
```
I think we need an additional test to cover this requirement: 
>This should be wrapped in a section html element with a role region attribute which is labelled via an aria-labelledby attribute which maps to the h2

Specifically the `labelledby` attribute part

mykeels replied:
Added clearer tests

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,99 @@
+          dir={dir}
+          labelId="features-analysis-heading"
+        >
+          Featured &amp; Analysis {/** this should be translated */}
```
```suggestion
          Features &amp; Analysis {/** this should be translated */}
```
Also are the translations not part of this PR?

mykeels replied:
Note: Check with @lizcameron

mykeels replied:
This is being tracked in https://github.com/bbc/simorgh/issues/5794, which I have assigned to myself, and intend to complete in conjunction with https://github.com/bbc/simorgh/issues/5765

andrewscfc replied:
Thanks for addressing the translation point, but placeholder text still has a typo we can correct for now 🙂

greenc05 commented:
https://github.com/bbc/simorgh/pull/5751#issuecomment-594460752 UX have confirmed that the background colour for the secondary column components is incorrect, this should be lunar (#f2f2f2) and will address the colour contrast issues.

Commit:
fix: use func prop

Commit:
chore: update snapshots

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
fix: rm grid wrapper

Commit:
fix: lint issues in test file

Commit:
test: add tests

- should have a section[role="region"][aria-labelledby]
- should have an [id] #features-analysis-heading

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

mykeels commented:
### [Preview from March 04, 2020]

After:
- overriding the background color of the H2 title
- changing the H3 links to DIV

---

![image](https://user-images.githubusercontent.com/11996508/75894749-f9c57880-5e34-11ea-8ae2-1168d8b3ea98.png)

Commit:
fix: change sec. column bg color to lunar

mykeels commented:
### [Preview from March 05, 2020]

After:
- Changing the Secondary Columns' background color from C_GHOST to C_LUNAR.

![image](https://user-images.githubusercontent.com/11996508/75973829-2087bb80-5ed6-11ea-9b54-ea5b483a5baa.png)

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

lizcameron commented:
Confirmation from UX that there should be no background colour for the Features and Analysis component.

mykeels commented on src/app/pages/StoryPage/index.jsx:
```diff
@@ -100,7 +101,7 @@ const StoryPage = ({ pageData }) => {
 
   const ComponentWrapper = styled.div`
-    background: ${C_CHALK};
+    background: ${C_LUNAR};
```
Remove this line, since the bg should be transparent.

Commit:
fix: rm gray background

Commit:
chore: revert to use Headline

Commit:
Merge branch 'latest' into add-features-analysis

mykeels commented:
## [Preview from March 06, 2020]

After:

- Reverting Headling to H3
- Removing background from Secondary Columns

Note: the Headline now has a line across it. That's from the `Bar` component in `@bbc/psammead--section-label`. There is reason to believe it has always been there, and the background made it hard to see.

![image](https://user-images.githubusercontent.com/11996508/76089428-f0194d80-5fb9-11ea-93c7-e4d22c893011.png)

HarveyPeachey commented:
> Note: the Headline now has a line across it. That's from the `Bar` component in `@bbc/psammead--section-label`. There is reason to believe it has always been there, and the background made it hard to see.
> 

The bar is intentional. The section label should have a background colour which is the same colour as the foreground document. You might be overriding it by accident maybe? 

This is what it should look like:
<img width="326" alt="Screenshot 2020-03-06 at 14 03 09" src="https://user-images.githubusercontent.com/4706195/76090332-48008600-5fb3-11ea-9bb6-d836c53a0927.png">

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

simonsinclair committed:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

Commit:
feat: make the header's text bg same as document body

Commit:
chore: update snapshots

Commit:
Merge branch 'add-features-analysis' of github.com:bbc/simorgh into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

mykeels commented:
### [Preview from March 09, 2020]

- Make the Features & Analysis Header text have the same bg as the document's body.

![image](https://user-images.githubusercontent.com/11996508/76197167-ec6d0d00-61eb-11ea-8fcc-c5f23e04ff12.png)

Commit:
chore: update bundle size

Commit:
fix: 1488 vulnerability

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

mykeels committed:
fix: use C_GHOST over C_WHITE

mykeels committed:
chore: update jest snapshot

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
chore: update snapshots

Commit:
fix: npm vulnerabilities in "kind--of"

Commit:
Merge branch 'latest' into add-features-analysis

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+  }
+
+  span[dir] {
+    background-color: ${C_GHOST};
```
Why does the background colour change dependent on the presence of a `dir` attribute?

mykeels replied:
The DOM structure of the Secondary Column Header is:

```html
<h2>
  <span>
    <span dir="...">...</span>
  </span>
</h2>
```

By default the `h2 span` is a gray color, which we do not need, so we make it transparent.

```
span {
    background-color: transparent;
  }
```

But then, there's a gray bar that runs through the text to the right. We need the actual text's background to be white, so the bar is just visible after it. 

That's why we use `span[dir]` to override the bg color for just the span that contains text.

andrewscfc replied:
Oh okay, so its a bit of a work around the Secondary Column Header styling, just not that obvious from looking at it. Not sure what the best way to make this clearer is, kinda feel like there should be a prop on `SectionLabel` to make it's background transparent but not sure that's worth the effort tbh.

andrewscfc commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+};
+
+FeaturesAnalysis.defaultProps = {
+  content: featuresAnalysis, // @TODO: rm this
```
Is todo before this is merged?

mykeels replied:
No, this is to be done in the next PR, that connects the features&analysis data to this component.

andrewscfc reviewed (commented):
Few more comments

Commit:
Merge branch 'latest' into add-features-analysis

andrewscfc reviewed (approved):
Thanks for going through my comments 🙂

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

thekp committed:
Merge branch 'latest' into add-features-analysis

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+          dir={dir}
+          labelId="features-analysis-heading"
+        >
+          Featured &amp; Analysis {/** this should be translated */}
```
- Can we remove this comment? They can lead to strange behaviour in JSX.
- "Featured" should be corrected to "Features" as per @andrewscfc's review.

mykeels replied:
![image](https://user-images.githubusercontent.com/11996508/76404793-5f5ebb00-6387-11ea-8298-87c6c36f6674.png) 😄

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.stories.jsx:
```diff
@@ -0,0 +1,63 @@
+      <RequestContextProvider
+        bbcOrigin="https://www.test.bbc.com"
+        isAmp={platform === 'amp'}
+        pageType="STY" /* Can also be one of other CPS pagetypes */
```
I'm going to ignore these comments as they're in a Story. 😄

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+};
+
+FeaturesAnalysis.defaultProps = {
+  content: featuresAnalysis, // @TODO: rm this
```
What does this comment mean?

mykeels replied:
It means `remove this`. When we switch from using fixture data to using actual frontpage data, the prop should default to an empty array instead.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+FeaturesAnalysis.propTypes = {
+  // We Reuse the front page story item blocks
+  // Both pages use CPS, so the data schema is the same
+  // This can be found under Frontpage data payloads: content.groups.find(group => group.type === 'feature-main').items
```
Is this comment necessary? It's clear from the code that's written we're reusing `storyItem` which are propTypes imported from `#models/propTypes/storyItem`.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+
+  span[dir] {
+    background-color: ${C_GHOST};
+  }
```
Why does omitting these styles have no effect? Are they unneeded now that the secondary column doesn't have a grey background?

mykeels replied:
They were added after the fact, actually. See my explanation to @andrewscfc https://github.com/bbc/simorgh/pull/5751#discussion_r390141884

mykeels replied:
You are right. They seem to be redundant. I've removed them.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+    margin-bottom: ${GEL_SPACING_TRPL};
+  }
+  /* z-index needs explicitly set as the psammead section label component uses negative z-indices */
+  z-index: 0;
```
I'm still unsure why a z-index is being set here. Please can you clarify why this is needed?

mykeels replied:
This was copied along with the code from `CpsRelatedContent` container. I'm not sure that it's necessary for this component though.

mykeels replied:
I've removed the `z-index`.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+import StoryPromo from '../StoryPromo';
+
+const Wrapper = styled(GridItemConstrainedLarge)`
+  margin-bottom: ${GEL_SPACING_DBL};
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    margin-bottom: ${GEL_SPACING_TRPL};
+  }
```
- Why are these rules necessary? The secondary column component should take care of margins.
- Also please review the content on mobile, it is overly compressed. You may need to reset some margins and padding there.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+import React, { useContext } from 'react';
+import { arrayOf, shape, node } from 'prop-types';
+import SectionLabel from '@bbc/psammead-section-label';
+import { C_GHOST } from '@bbc/psammead-styles/colours';
```
This is an unused import.

simonsinclair commented on src/app/containers/FeaturesAnalysis/index.jsx:
```diff
@@ -0,0 +1,90 @@
+    role: 'region',
+    'aria-labelledby': 'features-analysis-heading',
+  };
+  const FeaturesAnalysisWrapper = ({ children }) => (
+    <Wrapper {...a11yAttributes}>{children}</Wrapper>
+  );
```
Why does this warrant a new component? Can we set `a11yAttributes` on line 56?

mykeels replied:
This style was borrowed from `CpsRelatedContent` container.

In that case, the relevant code is:

```jsx
const RelatedContentWrapper = ({ children }) =>
    enableGridWrapper ? (
      <GhostGrid {...a11yAttributes}>
        <Wrapper>{children}</Wrapper>
      </GhostGrid>
    ) : (
      <Wrapper {...a11yAttributes}>{children}</Wrapper>
    );
```

It's clear that it needs to decide between wrapping with `<GhostGrid />` or not, but it's not clear why `<Wrapper />`, which the `RelatedContentWrapper` styles, is still included in its descendants. 

Perhaps, this has some effect on a11y I don't know about?

simonsinclair reviewed (commented):
This is a big piece of work, Michael. 👍 Please see my comments below.

Commit:
fix: secondary columns grid in IE

Commit:
chore: rm erring story comment

Commit:
fix: section label text

Commit:
chore: rm redundant comment

Commit:
fix: conditional padding for mobile

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
fix: rm redundant styles

Commit:
chore: update snapshots

HarveyPeachey committed:
Merge branch 'latest' into add-features-analysis

HarveyPeachey commented:
**Dev/A11y Swarm**

-https://bbc.github.io/accessibility-news-and-you/accessibility-news-and-testers
  - [x] Axe
  - [x] Design
  - [x] Colour Contrast
  - [x] Colour blindness
  - [x] Windows HCM
  - [x] FF Colour Preferences
  - [x] Text 200%, Page 200%
  - [x] Icons
  - [ ] ~Heading Order~
  - [x] Check accesibility of browser DOM
  - [x] Keyboard Navigation 

**_Screen Readers (IE11) - Harvey/Simon T:_** 
- [x] ZoomText Magnifier/Reader (Steps)
8. When hovering over the image it doesn't read out anything, need to confirm if this should be the case.
9. Speakit Tool reads out all content, also reads out the image alt text.
10. Similar to the speakit tool where it reads out all content and image alt text.
- [ ] Dragon NaturallySpeaking
- [x] JAWS (Steps)
4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent.
After navigating past one link JAWS pronounces "BLANK"
Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case.
- [x] Read&Write (Steps) 
4. "Read by sentence" setting read out both the link text and the hidden summary.
6. Same behaviour where it reads out the summary.
9. The feature is using a Faux block link so it cannot read out the img alt text.
- [x] NVDA (Firefox) - When clicking before the feature it doesn't go to the desired region. 
4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. The first link in the example infinitely loops speech when navigated to.
5. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. 
6. Same bug where it infinitely loops on the first link.
7.  "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.


**_Mac, iOS, and Android:_** 
- [x] VoiceOver iOS (iPhone/iPad)
5. Only on first image, it reads out alt text, then in english it says "actions availible very sharp - five faces"
8. "Features & Analysis Region" is read out in Spanish then it reads out "Heading level 2 - Features and Analysis" in Spanish.
- [ ] VoiceOver OS (Mac)
- [x] TalkBack (Chrome on Android)

HarveyPeachey committed:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

HarveyPeachey committed:
Merge branch 'latest' into add-features-analysis

greenc05 commented on src/app/containers/FeaturesAnalysis/__snapshots__/index.test.jsx.snap:
```diff
@@ -0,0 +1,143 @@
+                  STY - Ọrụ bekee na ịrụrụ onwe gị ọrụ, kedụ nk?
+                </a>
+              </h3>
+              <p
```
Lets remove this p tag and it's content please, this would not be there.

greenc05 commented:
> **Dev/A11y Swarm**
> 
> -https://bbc.github.io/accessibility-news-and-you/accessibility-news-and-testers
> 
> * [x]  Axe
> * [x]  Design
> * [x]  Colour Contrast
> * [x]  Colour blindness
> * [x]  Windows HCM
> * [x]  FF Colour Preferences
> * [x]  Text 200%, Page 200%
> * [x]  Icons
> * [ ]  ~Heading Order~
> * [x]  Check accesibility of browser DOM
> * [x]  Keyboard Navigation
> 
> **_Screen Readers (IE11) - Harvey/Simon T:_**
> 
> * [x]  ZoomText Magnifier/Reader (Steps)
> 
> 1. When hovering over the image it doesn't read out anything, need to confirm if this should be the case.
> 2. Speakit Tool reads out all content, also reads out the image alt text.
> 3. Similar to the speakit tool where it reads out all content and image alt text.
> 
> * [ ]  Dragon NaturallySpeaking
> * [x]  JAWS (Steps)
> 
> 1. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent.
>    After navigating past one link JAWS pronounces "BLANK"
>    Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case.
> 
> * [x]  Read&Write (Steps)
> 
> 1. "Read by sentence" setting read out both the link text and the hidden summary.
> 2. Same behaviour where it reads out the summary.
> 3. The feature is using a Faux block link so it cannot read out the img alt text.
> 
> * [x]  NVDA (Firefox) - When clicking before the feature it doesn't go to the desired region.
> 
> 1. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation. The first link in the example infinitely loops speech when navigated to.
> 2. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.
> 3. Same bug where it infinitely loops on the first link.
> 4. "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish translation.
> 
> **_Mac, iOS, and Android:_**
> 
> * [ ]  VoiceOver iOS (iPhone/iPad)
> * [ ]  VoiceOver OS (Mac)
> * [ ]  TalkBack (Chrome on Android)

**ZoomText Magnifier/Reader**
"8. When hovering over the image it doesn't read out anything, need to confirm if this should be the case." @HarveyPeachey This is due to faux block link CSS and expected

**JAWS**
"Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent." @HarveyPeachey were you testing this on Mundo?

"After navigating past one link JAWS pronounces "BLANK" - Fine

"Summary is being rendered in HTML but not being read out and is also hidden on the page - should this be the case." Why are we rendering the summary @mykeels? Is it possible not to please? This is also why it's reading out with Read&Write @HarveyPeachey 

**NVDA**
Sounds a bit buggy, might be advisable to re-test NVDA when we have the translations @HarveyPeachey ?

HarveyPeachey commented:
> 
> **JAWS**
> "Features & Analysis Region" is read out in English then it reads out "Heading level 2 - Features and Analysis" in a Spanish accent." @HarveyPeachey were you testing this on Mundo?
> 
> **NVDA**
> Sounds a bit buggy, might be advisable to re-test NVDA when we have the translations @HarveyPeachey ?

Yeah, I was testing on Mundo as that's where our big Test asset lives. The main point of that statement was that it read out "Features & Analysis Region" and then it also readout "Heading level 2 - Features and Analysis", which was a bit off as it's repeating the component name.

With the NVDA stuff, I agree this should be tested when translations are added, I'll make this into a seperate issue :).

Commit:
Merge branch 'latest' into add-features-analysis

mykeels commented:
> "Why are we rendering the summary @mykeels? Is it possible not to please? This is also why it's reading out with Read&Write @HarveyPeachey"

On a medium-sized screen, the summary is displayed. 
![image](https://user-images.githubusercontent.com/11996508/76770357-1f8c3f00-679e-11ea-9e51-39d5fd4881e3.png)

Commit:
chore: add timestamps back

following a11y bug talk with liz, and greenc01

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
chore: update bundle size config

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
chore: update bundle size

FK78 committed:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
chore: npm i

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
chore: update bundle sizes

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis

paruchurisilpa committed:
Merge branch 'latest' into add-features-analysis

paruchurisilpa committed:
Merge branch 'latest' into add-features-analysis

paruchurisilpa committed:
Merge branch 'latest' into add-features-analysis

Commit:
Merge branch 'latest' into add-features-analysis
